src/rsserver/p3serverconfig.cc

```diff
@@ -28,70 +28,73 @@
 
 RsServerConfig *rsConfig = NULL;
 
-static const std::string pqih_ftr("PQIH_FTR");
+static constexpr char PQIH_FTR[] = "PQIH_FTR";
 
-#define DEFAULT_DOWNLOAD_KB_RATE       (10000.0)
-#define DEFAULT_UPLOAD_KB_RATE         (10000.0)
+static constexpr float DEFAULT_DOWNLOAD_KB_RATE = 10000.0;
+static constexpr float DEFAULT_UPLOAD_KB_RATE   = 10000.0;
 
-#define MIN_MINIMAL_RATE	(5.0)
+static constexpr float MIN_MINIMAL_RATE = 5.0;
 
 
 p3ServerConfig::p3ServerConfig(p3PeerMgr *peerMgr, p3LinkMgr *linkMgr, p3NetMgr *netMgr, pqihandler *pqih, p3GeneralConfig *genCfg)
-:configMtx("p3ServerConfig")
+    :  mPeerMgr(peerMgr), mLinkMgr(linkMgr), mNetMgr(netMgr), mPqiHandler(pqih)
+    , mGeneralConfig(genCfg)
+    , configMtx("p3ServerConfig")
+    , mUserLevel(RsConfigUserLvl::NEW) // START LEVEL
+    , mRateDownload(DEFAULT_DOWNLOAD_KB_RATE), mRateUpload(DEFAULT_UPLOAD_KB_RATE)
+    , mRateDownloadWhenIdle(DEFAULT_DOWNLOAD_KB_RATE), mRateUploadWhenIdle(DEFAULT_UPLOAD_KB_RATE)
+    , mIsIdle(false), mOpMode(RsOpMode::FULL)
 {
-	mPeerMgr = peerMgr;
-	mLinkMgr = linkMgr;
-	mNetMgr = netMgr;
-	mPqiHandler = pqih;
-
-	mGeneralConfig = genCfg;
-
 	RsStackMutex stack(configMtx); /******* LOCKED MUTEX *****/
-
-	mUserLevel = RsConfigUserLvl::NEW; /* START LEVEL */
-	mRateDownload =  DEFAULT_DOWNLOAD_KB_RATE;
-	mRateUpload = DEFAULT_UPLOAD_KB_RATE;
-
-	mOpMode = RsOpMode::FULL;
-
 	rsConfig = this;
```

Comment:
This is really terrible coding IMO (I mean initializing a global variable from within the constructor of one instance of that global variable). If you have 2 mins, move that to rsInit.cc where all rsSomthing objects are initialized.